src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -482,14 +482,9 @@ export class DataResourceDataProvider implements IGridDataProvider {
 		// request params for the serialization request (eg, saveFormat, filePath).
 		let provider = this.cellModel.notebookModel.context?.providerName;
 		if (!provider) {
-			// If no connection currently exists, then pick the first connection provider for the current kernel.
-			// If there's still no provider, then fallback to the default MSSQL one.
-			let connProviders = this.cellModel.notebookModel.getApplicableConnectionProviderIds(this.cellModel.notebookModel.selectedKernelDisplayName);
-			if (connProviders?.length > 0) {
-				provider = connProviders[0];
-			} else {
-				provider = mssqlProviderName;
-			}
+			// Serizalize notebook query results to file is agnostic of database engine since the data is already in the notebook.
+			// We can let the mssql provider handle it so that other providers don't have to implement it.
+			provider = mssqlProviderName;
```

Comment:
I feel like we should first check if the current provider supports serialization, and if it does use that. Then only fall back to MSSQL if it doesn't. There may be cases where certain providers want to serialize data in a specific way for some reason.

Reply:
that sounds good to me. implemented in most recent commit.